hq_superset/api.py

```diff
@@ -0,0 +1,47 @@
+import json
+from http import HTTPStatus
+
+from flask import request
+from flask_appbuilder.api import BaseApi
+from flask_appbuilder.baseviews import expose
+from superset.superset_typing import FlaskResponse
+from superset.views.base import (
+    handle_api_exception,
+    json_error_response,
+    json_success,
+)
+
+from .models import DataSetChange
+
+
+class DataSetChangeAPI(BaseApi):
+    """
+    Accepts changes to datasets from CommCare HQ data forwarding
+    """
+
+    MAX_REQUEST_LENGTH = 10 * 1024 * 1024  # reject JSON requests > 10MB
+
+    def __init__(self):
+        self.route_base = '/hq_webhook'
+        self.default_view = 'post_dataset_change'
+        super().__init__()
+
+    @expose('/change/', methods=('POST',))
+    @handle_api_exception
+    def post_dataset_change(self) -> FlaskResponse:
```

Comment:
@kaapstorm Quickly remind me again, is this endpoint going to be hit for every row change in a data source?

Reply:
This endpoint will be hit for every change for a doc_id, where a doc_id originally refers to a case or a form. Sometimes one doc_id can result in multiple rows, e.g. a data source definition that pulls out a question inside a repeat group.